src/shogun/mathematics/graph/ops/shogun/Add.h

```diff
@@ -55,7 +56,35 @@ namespace shogun
 					if (CPU_arch->has_avx512f())
 						add_kernel_implementation_avx512f<T>(input1, input2, output, size);
 					else if (CPU_arch->has_avx())
-						add_kernel_implementation_avx<T>(input1, input2, output, size);
+					{
+						size_t i = 0;
+						if (size > 32/sizeof(T))
+						{
+							size_t remainder = size % (32/sizeof(T));
+							for (;i<size-remainder; i+=32/sizeof(T))
+							{
+								if constexpr(std::is_same_v<float, T>)
+								{
+									const auto packet1 = Eigen::internal::ploadu<Eigen::internal::Packet8f>(static_cast<const T*>(input1)+i);
+									const auto packet2 = Eigen::internal::ploadu<Eigen::internal::Packet8f>(static_cast<const T*>(input2)+i);
+									Eigen::internal::Packet8f output_packet;
+									add_kernel_implementation_avx<T>((void*)&packet1, (void*)&packet2, (void*)&output_packet);
+									Eigen::internal::pstoreu((static_cast<float*>(output)+i), output_packet);
+								}
+								else
+								{
+									add_kernel_implementation_avx<T>(static_cast<const T*>(input1)+i, 
+										static_cast<const T*>(input2)+i, 
+										static_cast<T*>(output)+i);
+								}
+							}
+						}
+						while(i < size)
```

Comment:
this should be manually unrolled with some sort of dispatching `unroll_op<T, remainder>(func, input1, input2, output)`